dev-util/clion/clion-2025.1-r1.ebuild

```diff
@@ -0,0 +1,150 @@
+	net-print/cups
+	sys-apps/dbus
+	sys-libs/zlib
+	x11-libs/cairo
+	x11-libs/libdrm
+	x11-libs/libX11
+	x11-libs/libXcomposite
+	x11-libs/libXcursor
+	x11-libs/libXdamage
+	x11-libs/libXext
+	x11-libs/libXfixes
+	x11-libs/libXi
+	x11-libs/libXrandr
+	x11-libs/libXrender
+	x11-libs/libXtst
+	x11-libs/libXxf86vm
+	x11-libs/libxcb
+	x11-libs/libxkbcommon
+	x11-libs/pango"
+
+QA_PREBUILT="opt/${PN}/*"
+
+src_prepare() {
+	my_arch_suffix=$(usex amd64 x64 aarch64)
+	my_arch_radler=$(usex amd64 x64 arm64)
+	local other_arch_amd64_aarch64=$(usex amd64 aarch64 amd64)
+	local other_arch_radler=$(usex amd64 arm64 x64)
+	tc-export OBJCOPY
+	default
+
+	local remove_me=(
+		Install-Linux-tar.txt
+		help/ReferenceCardForMac.pdf
+		bin/cmake
+		bin/gdb/linux
+		bin/lldb/linux
+		bin/ninja
+		plugins/remote-dev-server/selfcontained
+	)
+	remove_me+=(
+		lib/async-profiler/${other_arch_amd64_aarch64}
+		plugins/clion-radler/DotFiles/linux-${other_arch_radler}
+		plugins/clion-radler/dotTrace.dotMemory/DotFiles/linux-${other_arch_radler}
+		plugins/python-ce/helpers/pydev/pydevd_attach_to_process/attach_linux_${other_arch_amd64_aarch64}.so
+	)
+	use !amd64 && remove_me+=(
+		plugins/python-ce/helpers/coveragepy_old/coverage/tracer.cpython-310-x86_64-linux-gnu.so
+		plugins/python-ce/helpers/pydev/pydevd_attach_to_process/attach_linux_x86.so
+	)
+
+	rm -rv "${remove_me[@]}" || die
+
+	# excepting files that should be kept for remote plugins
+	skip_nonx86_files=(
+		"plugins/platform-ijent-impl/ijent-$(usex amd64 aarch64 x86_64)-unknown-linux-musl-release"
+		"plugins/clion-radler/DotFiles/linux-musl-${other_arch_radler}/jb_zip_unarchiver"
+		"plugins/clion-radler/DotFiles/linux-arm/jb_zip_unarchiver"
+		"plugins/clion-radler/DotFiles/linux-musl-arm/jb_zip_unarchiver"
+		"plugins/gateway-plugin/lib/remote-dev-workers/remote-dev-worker-linux-$(usex amd64 arm64 amd64)"
+	)
+	# removing debug symbols and relocating debug files as per #876295
+	# we're escaping all the files that contain $() in their name
+	# as they should not be executed
+	find . -type f ! -name '*$(*)*' -print0 | while IFS= read -r -d '' file; do
+		for skip in "${skip_nonx86_files[@]}"; do
+			[[ ${file} == "./${skip}" ]] && continue 2
+		done
+		if file "${file}" | grep -qE "ELF (32|64)-bit"; then
+			${OBJCOPY} --remove-section .note.gnu.build-id "${file}" || die
+			debugedit -b "${EPREFIX}/opt/${PN}" -d "/usr/lib/debug" -i "${file}" || die
+		fi
+	done
+
+	patchelf --set-rpath '$ORIGIN' "jbr/lib/libjcef.so" || die
+	patchelf --set-rpath '$ORIGIN' "jbr/lib/jcef_helper" || die
+	patchelf --set-rpath '$ORIGIN/../lib' "bin/clang/linux/${my_arch_suffix}/lib/libclazyPlugin.so" || die
+}
+
+src_install() {
+	local dir="/opt/${PN}"
+
+	insinto "${dir}"
+	doins -r *
+	fperms 755 "${dir}"/bin/{clion.sh,format.sh,fsnotifier,inspect.sh,jetbrains_client.sh,ltedit.sh,remote-dev-server.sh,restarter,clang/linux/${my_arch_suffix}/bin/{clangd,clang-tidy,clazy-standalone,llvm-symbolizer}}
+
+	if [[ -d jbr ]]; then
```

Comment:
Why is this here?  Is the `jbr` directory missing on one of the arches?

Reply:
Not a part of the arm64 commit diff, just kept as is

Reply:
> Why is this here? Is the `jbr` directory missing on one of the arches?

It's a old issue, most probably due to bundled vs unbundled jdk.